src/rust/ide/src/double_representation.rs

```diff
@@ -24,3 +27,62 @@ pub mod test_utils;
 ///
 /// Link: https://github.com/luna/enso/blob/main/doc/syntax/encoding.md
 pub const INDENT : usize = 4;
+
+
+
+// ====================
+// === ReferentName ===
+// ====================
+
+
+// === Errors ===
+
+/// Happens if a given string does not fulfill requirements of the referent name;
+#[derive(Clone,Debug,Fail)]
+#[fail(display="The `{}` is not a valid referent name.",_0)]
+pub struct NotReferentName(String);
+
+
+// === Definition ===
+
+#[derive(Clone,Debug,Display,Shrinkwrap,PartialEq,Eq,Hash)]
+/// The name segment is a string that starts with an upper-cased character.
+///
+/// It is used for naming modules, module path segments and projects.
+///
+/// This value corresponds to contents of the `Cons` AST shape.
+pub struct ReferentName(String);
+
+impl ReferentName {
+    /// Check if the given text would be a valid referent name;
+    pub fn validate(name:impl AsRef<str>) -> Result<(), NotReferentName> {
+        let name = name.as_ref();
+        // TODO [mwu]
+        //  We should be able to call parser or sth to verify that other requirements for the
+        //  referent form identifiers are fulfilled.
```

Comment:
When we'll do it? Add an info, that we could do it when rust parser will be available.